### PR TITLE
add canonical url to 2.3 to direct google to latest

### PR DESCRIPTION
**Pull request by samccann**
##### SUMMARY

We are still getting some top google search hits to 2.3 docs. This is the first cut attempt to teach search engines to use /latest/ instead.

##### ISSUE TYPE

- Docs Pull Request

##### COMPONENT NAME

docs/docsite/_themes/srtd/layout.html
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: docs/docsite/_themes/srtd/layout.html [hunk 1]
```diff
@@ -38,6 +38,9 @@
   <link href='https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700' rel='stylesheet' type='text/css'>
   <link href='https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.0.3/css/font-awesome.min.css' rel='stylesheet' type='text/css'>
 
+  {# CANONICAL URL #}
+  <link rel="canonical" href="https://docs.ansible.com/ansible/latest/index.html">
+
```
Comment (tremble) [inline docs/docsite/_themes/srtd/layout.html]: Is this not going to result in **all** 2.3 results referencing the index page for latest, including all module docs?

Will these pages be "near identical" enough that Google et al will actually default to the "latest" versions?

Comment (samccann) [reply docs/docsite/_themes/srtd/layout.html]: dang you're right. That's what I get for thinking I had the magic solution right before I log off for the day. The newer releases have support for canonical_url in the sphinx theme but I couldn't get that to work here. I'll dig further into the theme changes to find out how to add it there so it appends the page html correctly.

This isn't the magic complete solution for sure. The hope is that some of the pages are close enough that google will redirect to latest, but yeah, the further the pages drift, the less helpful this is.

The complete solution will come next - creating a docs archive site and moving the EOL docs there.

Comment (samccann) [reply docs/docsite/_themes/srtd/layout.html]: found how this was solved in 2.4 https://github.com/ansible/ansible/pull/53801 so using that cherry-pick instead. seems to work.

**Comment by samccann** [comment]
Verified this adds the canonical urls correctly now:
image

And currently staged at http://docs.testing.ansible.com/ansible/2.3/

Since this is an ancient branch and requires 6 approvals, we may have to republish 2.3 from this PR branch?

**Pull request closed without merging**